### DISCUSSION TITLE
Getting an authorization token

### DISCUSSION THREAD

**Discussion by analyserdmz** [directus/directus#2757] [Archive]
I am trying to get a token by sending my POST data (mail & pass) to the API but no luck so far.

My project's name is slotspanel, and my token request test script is below.

```
$data = array(
	'email'      => 'my-Email-Here@mail.com',
	'password'    => 'MyPasswordHere'
);

$options = array(
	'http' => array(
		'method'  => 'POST',
		'content' => json_encode( $data ),
		'header'=>  "Content-Type: application/json\r\n" .
					"Accept: application/json\r\n"
	)
);

$context  = stream_context_create( $options );
$result = file_get_contents("http://127.0.0.1/slotspanel/auth/authenticate", false, $context);
$response = json_decode($result);

echo $response;
```

The response I get, instead of getting a JSON object, is...
`
Warning: file_get_contents(http://127.0.0.1/slotspanel/api/auth/authenticate): failed to open stream: HTTP request failed! HTTP/1.1 401 Unauthorized in C:\xampp_new\htdocs\index2.php on line 18
`
Do I do anything wrong here?

**Comment by rijkvanzanten** [comment]
The error indicates that you were trying to open `http://127.0.0.1/slotspanel/api/auth/authenticate`, but your code example says `http://127.0.0.1/slotspanel/api/auth/authenticate`. Are you sure the server is running under `127.0.0.1/slotspanel/api/`? Are you using Apache? Is mod_rewrite working?

**Reply by analyserdmz** [reply]
@rijkvanzanten i've downloaded the latest (version 8) release. Should I install the API extra? Isn't it included already?

**Comment by analyserdmz** [comment]
@rijkvanzanten I'm sorry about the difference in the script and the error. I was trying different solutions found on web and it seems my code was copied after changing the path back to original.

To your question, yes, I am sure the full path is http://127.0.0.1/slotspanel (and not /api/), and I am also using a XAMPP Server (latest PHP 7) for development purposes on windows.

Any suggestion will be appreciated.

EDIT:
**LoadModule rewrite_module modules/mod_rewrite.so** is uncommented and turned **AllowOverride None to AllowOverride All** but with no luck. I still get the same error.


Error log entry for each refresh I do on my script
```
[2020-06-18 13:36:29] api[auth].ERROR: Directus\Exception\UnauthorizedException: Unauthorized request in C:\xampp_new\htdocs\src\helpers\app.php:287
Stack trace:
#0 C:\xampp_new\htdocs\vendor\slim\slim\Slim\App.php(518): Directus\{closure}
#1 C:\xampp_new\htdocs\src\core\Directus\Application\Http\Middleware\CorsMiddleware.php(71): Slim\App->__invoke
#2 [internal function]: Directus\Application\Http\Middleware\CorsMiddleware->__invoke
#3 C:\xampp_new\htdocs\vendor\slim\slim\Slim\DeferredCallable.php(57): call_user_func_array
#4 [internal function]: Slim\DeferredCallable->__invoke
#5 C:\xampp_new\htdocs\vendor\slim\slim\Slim\MiddlewareAwareTrait.php(70): call_user_func
#6 C:\xampp_new\htdocs\vendor\slim\slim\Slim\MiddlewareAwareTrait.php(117): Slim\App->Slim\{closure}
#7 C:\xampp_new\htdocs\vendor\slim\slim\Slim\App.php(392): Slim\App->callMiddlewareStack
#8 C:\xampp_new\htdocs\vendor\slim\slim\Slim\App.php(297): Slim\App->process
#9 C:\xampp_new\htdocs\src\core\Directus\Application\Application.php(161): Slim\App->run
#10 C:\xampp_new\htdocs\slotspanel\index.php(5): Directus\Application\Application->run [] []

```

**Reply by rijkvanzanten** [reply]
Is `slotspanel` the name of your Directus folder or the name of your project in Directus?

**Reply by analyserdmz** [reply]
> 
> 
> Is `slotspanel` the name of your Directus folder or the name of your project in Directus?

Uhm, both.

**Reply by rijkvanzanten** [reply]
The `/` route should redirect to `/admin`, while `/<unknown project>` will in fact give an unauthorized error:

Redirects: https://demo.directus.io 
Wrong project: https://demo.directus.io/non-existing/
Existing project: https://demo.directus.io/thumper/

I'm starting to think your project might not be named `slotspanel`, which would explain the not-authorized message. What's the name of your config file under the `config` directory in the API?

**Reply by analyserdmz** [reply]
- http://127.0.0.1/slotspanel redirects to http://127.0.0.1/slotspanel/admin
- My project name is the same as the folder name (should i change the project name?)

**Reply by analyserdmz** [reply]
Name of the config: slotspanel.php

**Comment by analyserdmz** [comment]
@rijkvanzanten never mind. I had to change my URL to http://127.0.0.1/slotspanel/slotspanel/auth/authenticate where the first /slotspanel is the directory and the second is the project name. Thank you for your time!
